bundle/src/main/java/com/adobe/acs/commons/rewriter/impl/ContentVariableTransformer.java

```diff
@@ -0,0 +1,106 @@
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * {@link org.apache.sling.rewriter.Transformer} used to process HTML requests and replace content tokens found in the
+ * rendered HTML.
+ */
+public class ContentVariableTransformer extends ContentHandlerBasedTransformer {
+
+    private Map<String, Object> contentVariableReplacements;
+    private PropertyAggregatorService aggregatorService;
+
+    public ContentVariableTransformer() {
+    }
+
+    public ContentVariableTransformer(PropertyAggregatorService propertyAggregatorService) {
+        this.aggregatorService = propertyAggregatorService;
+    }
+
+    @Override
+    public void init(ProcessingContext processingContext, ProcessingComponentConfiguration processingComponentConfiguration) throws IOException {
+        SlingHttpServletRequest request = processingContext.getRequest();
+
+        contentVariableReplacements = aggregatorService.getProperties(request);
+    }
+
+    public void startElement(String uri, String localName, String quaName, Attributes atts) throws SAXException {
+        if (shouldRun() && localName.equals("a")) {
+            AttributesImpl newAttrs = new AttributesImpl(atts);
+            for (int i = 0; i < newAttrs.getLength(); i++) {
+                String currentAttribute = newAttrs.getValue(i);
+                final List<String> keys = ContentVariableReplacementUtil.getKeys(currentAttribute);
+                for (String key : keys) {
+                    // If the placeholder key is in the map then replace it
+                    if (contentVariableReplacements.containsKey(key)) {
+                        String replaceValue = (String) contentVariableReplacements.get(key);
+                        newAttrs.setValue(i, currentAttribute.replace(ContentVariableReplacementUtil.getPlaceholder(key), replaceValue));
+                    }
+                }
+            }
+            getContentHandler().startElement(uri, localName, quaName, newAttrs);
+        } else {
+            getContentHandler().startElement(uri, localName, quaName, atts);
+        }
+    }
+
+    public void characters(char[] ch, int start, int length) throws SAXException {
+        String currentString = new String(ch);
+        int chunkLength = length;
+
+        if (shouldRun()) {
+            String currentChunk = currentString.substring(start, start + length);
+
+            // Get the current placeholders in the string
+            final List<String> keys = ContentVariableReplacementUtil.getKeys(currentChunk);
+            for (String key : keys) {
+                // If the placeholder key is in the map then replace it
+                if (contentVariableReplacements.containsKey(key)) {
+                    final String placeholderReplacement = String.valueOf(contentVariableReplacements.get(key));
+                    final String replacedChunk = currentChunk.replace(ContentVariableReplacementUtil.getPlaceholder(key), placeholderReplacement);
```

Comment:
Can we rename this to `chunkWithReplacements`?  `replacedChunk` has been confusing me when reading this code - in my mind the "replaced chunk" is the "key" (the characters that we replaced)